tests/e2e/specs/setup-mc/step-1-accounts.test.js

```diff
@@ -397,26 +410,15 @@ test.describe( 'Set up accounts', () => {
 			expect( url ).toMatch( /^https:\/\/example\.com(\/|\?|$)/ );
 		} );
 
-		test.describe( 'Account claimed', () => {
-			test.beforeEach( async () => {
-				await setUpAccountsPage.mockAdsStatusClaimed();
-				await setUpAccountsPage.mockAdsAccountConnected();
-
-				await setUpAccountsPage.goto();
-			} );
+		test( 'should see the merchant center id once account has been claimed', async () => {
```

Comment:
This will pass, but doesn't really test that the claim was successful because the MC ID is shown even when the claim account message is visible. I think a better test would wait for the UI to update and confirm that after we receive a successful claim response the UI looks as intended.

Here's an idea that seems to work, though I'm not sure the focus action is working as expected because it take a long time for the account-status polling function to kick in:


```diff
diff --git a/tests/e2e/specs/setup-mc/step-1-accounts.test.js b/tests/e2e/specs/setup-mc/step-1-accounts.test.js
index 641bfce68..51cc5fbed 100644
--- a/tests/e2e/specs/setup-mc/step-1-accounts.test.js
+++ b/tests/e2e/specs/setup-mc/step-1-accounts.test.js
@@ -408,17 +408,34 @@ test.describe( 'Set up accounts', () => {
 			await popupPage.waitForLoadState();
 			const url = popupPage.url();
 			expect( url ).toMatch( /^https:\/\/example\.com(\/|\?|$)/ );
+
+			await popupPage.close();
 		} );
 
 		test( 'should see the merchant center id once account has been claimed', async () => {
 			await setUpAccountsPage.mockAdsStatusClaimed();
 			await setUpAccountsPage.mockAdsAccountConnected();
 
-			await setUpAccountsPage.goto();
+			const statusResponse =
+				setUpAccountsPage.registerAdsAccountStatusResponse();
+
+			await setUpAccountsPage.page.dispatchEvent( 'body', 'focus' );
+
+			await statusResponse;
+
 			const googleAccountCard = setUpAccountsPage.getGoogleAccountCard();
 			await expect( googleAccountCard ).toContainText(
 				'Google Ads ID: 12345'
 			);
+
+			await expect(
+				googleAccountCard.getByText(
+					'Google Ads conversion measurement has been set up for your store.',
+					{
+						exact: true,
+					}
+				)
+			).toBeVisible();
 		} );
 	} );
 
diff --git a/tests/e2e/utils/pages/setup-mc/step-1-set-up-accounts.js b/tests/e2e/utils/pages/setup-mc/step-1-set-up-accounts.js
index cf50408e4..be68be31f 100644
--- a/tests/e2e/utils/pages/setup-mc/step-1-set-up-accounts.js
+++ b/tests/e2e/utils/pages/setup-mc/step-1-set-up-accounts.js
@@ -366,6 +366,19 @@ export default class SetUpAccountsPage extends MockRequests {
 		);
 	}
 
+	/**
+	 * Register the responses when checking for account-status.
+	 *
+	 * @return {Promise<import('@playwright/test').Response>} The response.
+	 */
+	registerAdsAccountStatusResponse() {
+		return this.page.waitForResponse(
+			( response ) =>
+				response.url().includes( '/gla/ads/account-status' ) &&
+				response.status() === 200
+		);
+	}
+
 	/**
 	 * Get terms checkbox.
 	 *
```

Reply:
Thanks @joemcgill . I've updated the test. Can you kindly take a look now?